internal/ir/xds.go

```diff
@@ -1683,6 +1683,8 @@ const (
 // DestinationEndpoint holds the endpoint details associated with the destination
 // +kubebuilder:object:generate=true
 type DestinationEndpoint struct {
+	// Hostname refers to the endpoint's hostname
+	Hostname *string `json:"hostname,omitempty" yaml:"hostname,omitempty"`
```

Comment:
should this be per DestinationSetting instead ?

Reply:
although in xds, its per endpoint, from the input API perspective, its per Setting, by doing this we also save space for cases with many endpoints

Reply:
Done, I rewrote it in DestinationSetting.